### ISSUE TITLE
HTTP request hangs when sending empty file

### ISSUE THREAD

**Issue by sergeyminaev** [clj-commons/aleph#559]
```clojure
  @(aleph.http/put "https://localhost:3333"
                   {:body (clojure.java.io/file "empty.txt")
                    :pool (aleph.http/connection-pool {:connection-options {:ssl-context ctx}})
                    :request-timeout 1000})
```

This code throws TimeoutException because of converting empty file to Manifold source inside `aleph.http.core` fails with NullPointerException.

**Comment by kachayev** [comment]
@sergeyminaev Thanks for reporting! 1.0.0 branch has a few improvements for errors handling, a few more are introduces in [this](https://github.com/clj-commons/aleph/pull/553) PR. I  believe that this specific issue were addressed there (and there's a unit test to cover it).

**Comment by KingMob** [comment]
Fixed by #590
